src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -275,25 +448,217 @@ public void updateNonExistingEventActivity() throws Exception {
 
         int databaseSizeBeforeDelete = eventActivityRepository.findAll().size();
 
-        // Delete the eventActivity
         restEventActivityMockMvc.perform(delete("/api/event-activities/{id}", eventActivity.getId())
             .accept(TestUtil.APPLICATION_JSON_UTF8))
             .andExpect(status().isNoContent());
 
-        // Validate the database contains one less item
         List<EventActivity> eventActivityList = eventActivityRepository.findAll();
         assertThat(eventActivityList).hasSize(databaseSizeBeforeDelete - 1);
     }
 
     @Test
-    @Transactional
+    public void deleteEventActivity_WithEventActivityNotExist_ShouldThrow400() throws Exception {
+
+        int databaseSizeBeforeDelete = eventActivityRepository.findAll().size();
+
+        restEventActivityMockMvc.perform(delete("/api/event-activities/{id}", Long.MAX_VALUE)
+            .accept(TestUtil.APPLICATION_JSON_UTF8))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeDelete);
+    }
+
+    @Test
+    public void deleteEventActivity_WithEventIsCancelled_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CANCELLED);
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeDelete = eventActivityRepository.findAll().size();
+
+        assertThat(savedEvent.getStatus()).isEqualByComparingTo(EventStatus.CANCELLED);
+
+        restEventActivityMockMvc.perform(delete("/api/event-activities/{id}", savedEventActivity.getId())
+            .accept(TestUtil.APPLICATION_JSON_UTF8))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeDelete);
+    }
+
+    @Test
+    public void deleteEventActivity_WithEventIsClosed_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CLOSED);
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeDelete = eventActivityRepository.findAll().size();
+
+        assertThat(savedEvent.getStatus()).isEqualByComparingTo(EventStatus.CLOSED);
+
+        restEventActivityMockMvc.perform(delete("/api/event-activities/{id}", savedEventActivity.getId())
+            .accept(TestUtil.APPLICATION_JSON_UTF8))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeDelete);
+    }
+
+    @Test
+    public void deleteEventActivity_WithEventIsEnded_ShouldThrow400() throws Exception {
+        Event savedEvent = initEventDB();
```

Comment:
when you set the end date of event to be end?

Reply:
fixed in 32c8feeefe1ca6221761c8748a79059953d83cda